WordPress/Classes/Services/EditorSettings.swift

```diff
@@ -67,4 +84,42 @@ class EditorSettings: NSObject {
             database.set(newValue, forKey: nativeEditorEnabledKey)
         }
     }
+
+    // We can't return a type that's both a PostEditor and a UIViewController, so using
+    // a configure block as a hack.
+    // In Swift 4, we'll be able to do `instantiateEditor() -> UIViewController & PostEditor`,
+    // and then let the caller configure the editor.
+    func instantiatePostEditor(post: AbstractPost, configure: (PostEditor, UIViewController) -> Void) -> UIViewController {
```

Comment:
It feels slightly off, but then all EditorSettings does so far is figure out which editor is enabled, so I think it fits: you configure it from the settings toggles, and as a result you can instantiate an editor of the right type.